orderer/common/multichannel/registrar.go

```diff
@@ -259,7 +278,7 @@ func (r *Registrar) initSystemChannel(existingChannels []string) {
 
 // initAppChannels initializes application channels, assuming that the system channel does NOT exist.
 // This implies that the orderer is using the channel participation API for joins (channel creation).
-func (r *Registrar) initAppChannels(existingChannels []string, channelsWithJoinBlock map[string]*cb.Block) {
+func (r *Registrar) initAppChannels(existingChannels []string, channelsWithJoinBlock map[string]*cb.Block, sysChannel string) {
```

Comment:
by this time r.systemChannelID signals whether we have a system channel or not.... and, read the comment:
"assuming that the system channel does NOT exist"